4-storage/storage-loader/lib/snowplow-storage-loader/dashdb_loader.rb

```diff
@@ -0,0 +1,122 @@
+          Monitoring::Snowplow.instance.track_load_failed('Failure to transfer atomic events from S3 => DashDB, response #{response_value}.')
+        end
+      end
+
+      if snowplow_tracking_enabled
+        Monitoring::Snowplow.instance.track_load_succeeded()
+      end
+
+      nil
+    end
+    module_function :load_atomic_events_into_dashdb
+
+    private
+
+      # Call dataworks_proxy api endpoint with a request body that passes DashDB and S3 details.
+      # The endpoint will create the S3 => DashDB activity and also kick it off.
+      # ** Still waiting on DataWorks team to add new functionality where file pattern matching is allowed.
+      #
+      # Parameters:
+      # +config+:: the configuration options
+      # +target+:: the configuration for this specific target
+      # +run_folder+:: the retrieved run folder from shredded/good
+      Contract Hash, Hash => Num
+      def create_run_dataworks_transfer(config, target, run_folder)
+        url = URI.parse(target[dataworks_proxy])
+        req = Net::HTTP::Post.new(url)
+        req['Accept'] = 'application/json'
+        req.body = '{
+          "api_key":"#{target[:api_key]}",
+          "host":"#{target[:host]}",
+          "username":"#{target[:username]}",
+          "password":"#{target[:password]}",
+          "database":"#{target[:database]}",
+          "schema":"#{target[:schema]}",
+          "table":"#{target[:table]}",
+          "s3_region":"#{config[:aws][:s3][:region]}",
+          "s3_access_key":"#{config[:aws][:access_key_id]}",
+          "s3_secret_key":"#{config[:aws][:secret_access_key]}",
+          "s3_bucket":"#{config[:aws][:s3][:buckets][:shredded][:good]}/#{run_folder}"
+        }'
+
+        res = Net::HTTP.start(url.host, url.port, :use_ssl => url.scheme == 'https') { |http| http.request(req) }
+
+        res.value
+      end
+      module_function :create_run_dataworks_transfer
+
+      # Retrieve run folder from shredded/good. Max one folder in there at any given time.
+      #
+      # Parameters:
+      # +config+:: the configuration options
+      Contract Hash => String
+      def retrieve_run_folder(config, snowplow_tracking_enabled)
+        s3 = Sluice::Storage::S3::new_fog_s3_from(
+          config[:aws][:s3][:region],
+          config[:aws][:access_key_id],
+          config[:aws][:secret_access_key])
+
+        loc = Sluice::Storage::S3::Location.new(config[:aws][:s3][:buckets][:shredded][:good])
+
+        shredded_good_run_filepath = Sluice::Storage::S3::list_files(s3, loc).find { |file| RUN_PATTERN.match(file.key) }
+
+        if shredded_good_run_filepath.nil?
+          if snowplow_tracking_enabled
+            Monitoring::Snowplow.instance.track_load_failed('Cannot find atomic-events directory in shredded/good')
+          end
+
+          raise DatabaseLoadError, 'Cannot find atomic-events directory in shredded/good'
+        end
+
+        altered_enriched_subdirectory = RUN_PATTERN.match(altered_enriched_filepath.key)[1]
```

Comment:
Do you need the `altered_enriched_subdirectory = ` there - could just return the right-hand-side of the assignment?